src/nautilus_librarian/mods/libvips/domain/process_image.py

```diff
@@ -0,0 +1,15 @@
+from nautilus_librarian.mods.libvips.domain.utils import get_image, resample_image, save_image
+import os
+
+def create_output_folder(destination_filename):
+    os.makedirs(os.path.dirname(destination_filename), exist_ok=True)
+
+def process_image(filename, destination_filename: str, base_image_size: int):
+    """
+    Generates and saves an image file using a source image, with the specified size
+    """
+    image = get_image(filename)
+    image = resample_image(image, base_image_size)
+    create_output_folder(destination_filename)
+    save_image(image, destination_filename)
```

Comment:
I'm not really sure if it is necessary, as the libvips methods are very simple, and I think that the code would remain very similar (in terms of extension and legibility) with this wrapper, compared with how it is now.
Anyway, it's something worth deeper discussion and, in case we go for it, can be implemented in a separate issue.